src/zenml/services/service.py

```diff
@@ -232,21 +290,28 @@ def update_status(self) -> None:
             "Running status check for service '%s' ...",
             self,
         )
-        state, err = self.check_status()
-        logger.debug(
-            "Status check results for service '%s': %s [%s]",
-            self,
-            state.name,
-            err,
-        )
-        self.status.update_state(state, err)
+        try:
+            state, err = self.check_status()
+            logger.debug(
+                "Status check results for service '%s': %s [%s]",
+                self,
+                state.name,
+                err,
+            )
+            self.status.update_state(state, err)
 
-        # don't bother checking the endpoint state if the service is not active
-        if self.status.state == ServiceState.INACTIVE:
-            return
+            # don't bother checking the endpoint state if the service is not active
+            if self.status.state == ServiceState.INACTIVE:
+                return
 
-        if self.endpoint:
-            self.endpoint.update_status()
+            if self.endpoint:
```

Comment:
Just out of curiosity, if the status of the service is inactive, should we also update the status of the endpoint as well? The same goes for the `except` block.

Reply:
I think for the current service implementation we don't do that since we don't look for service endpoint directly. but with the new update services we might wanna do that